index.html

```diff
@@ -1,100 +1,122 @@
+            var selectedOption = this.value;
+            if (loadedFile) {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
 
+                    displayResults(districtRows);
+                    kwadratodleglosci();
+                };
 
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV.');
+            }
+        });
 
+        function kwadratodleglosci() {
+            var selectedOption = document.getElementById('select').value;
+            if (loadedFile && selectedOption !== "") {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
+
+                    var squaredDistance = 0;
+                    for (var i = 0; i < districtRows.length; i++) {
+                    var rowData = districtRows[i].split(';');
+                    var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                    squaredDistance += Math.pow(votersCount, 2);
+                    }
+
+                    document.getElementById('kwadratodleglosci').innerHTML = 'Kwadrat odległości dla Okręgu Wyborczego ' + selectedDistrict + ': ' + squaredDistance;
+                };
+
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV i wybrać okręg.');
+            }
+        }
+
+        function populateDropdown() {
+            var reader = new FileReader();
+            reader.onload = function (e) {
+                var csvData = e.target.result;
+                var rows = csvData.split('\n');
+                var selectDropdown = document.getElementById('select');
+                selectDropdown.innerHTML = "";
+                addedDistricts = [];
+                for (var i = 0; i < rows.length; i++) {
+                    var districtNumber = rows[i].split(';')[0];
+                    if (!addedDistricts.includes(districtNumber)) {
+                    var optionValue = i;
+                    var optionText = "Okreg wyborczy " + districtNumber;
+                    var option = document.createElement("option");
+                    option.value = optionValue;
+                    option.text = optionText;

```

Comment:
Minor:

Instead of creating the `optionValue` and `optionText` variables, you can inline them, which ends up being 2 lines shorter, and easier to read:

```suggestion
                    var option = document.createElement("option");
                    option.value = i;
                    option.text = "Okreg wyborczy " + districtNumber;

```